src/seedu/addressbook/AddressBook.java

```diff
@@ -486,8 +497,16 @@ private static String getMessageForPersonsDisplayedSummary(ArrayList<String[]> p
         final ArrayList<String[]> matchedPersons = new ArrayList<>();
         for (String[] person : getAllPersonsInAddressBook()) {
             final Set<String> wordsInName = new HashSet<>(splitByWhitespace(getNameFromPerson(person)));
-            if (!Collections.disjoint(wordsInName, keywords)) {
-                matchedPersons.add(person);
+            Boolean hasMatchFound = false;
+            for(String keyword : keywords) {
+            	for(String wordInName : wordsInName) {
+            		if(keyword.equalsIgnoreCase(wordInName)) {
+            			hasMatchFound = true;
+            		}
+            	}
+            }
+            if(hasMatchFound) {
+            	matchedPersons.add(person);
             }
```

Comment:
When you modify the behavior, remember to update documentation and tests as well.